datalad/interface/common_cfg.py

```diff
@@ -80,6 +80,15 @@
         'destination': 'local',
         'type': bool,
     },
+    # this is actually used in downloaders, but kept cfg name original
+    'datalad.credential.force-ask': {
```

Comment:
Minor, but I feel it's somewhat strange/inconsistent to use `credential` instead of `credentials` in that name. No strong objection, though.
However, if I think ahead imagining that we may want to enhance this, addressing a particular set of credentials by a configuration, singular at that level feels even more strange ;-)

Reply:
If someone can provide the rules for switching between singular and plural, please write them down. Our use of that is a wild ride. I hardly ever (consciously) see use of plural outside datalad. I am not aware of a single instance of a Git config setting that uses plural (NB. Git has `credential. ...`).

I have been guilty of introducing the use of plural in the past myself. I regret all of them.

Reply:
TL;DR:  I think going for plural at this point in time would be more consistent with current state of affairs.

<details>
<summary>TL:</summary> 

- we have `datalad.externals`, `datalad.tests` and `datalad.locations` already... the singular ones kinda make sense singular 
<details>
<summary>full grep</summary> 

```shell
      1     'datalad.annex
      1     'datalad.crawl
      1     'datalad.exc
      1     'datalad.externals
      1     'datalad.github
      1     'datalad.install
      8     'datalad.locations
      9     'datalad.log
      4     'datalad.metadata
      3     'datalad.repo
      5     'datalad.runtime
      1     'datalad.save
      3     'datalad.search
      2     'datalad.ssh
     16     'datalad.tests
      4     'datalad.ui
      1                     "See 'datalad.ui
      1                'title': 'Include exceptions and their traceback in log messages. If set, \'datalad.exc

```
</details>

so it feels like `datalad.credentials` would be more consistent with that

- also `datalad.credential` somewhat suggests to have some ultimate singular credential, thus confusing
- and our commands ATM are plural, e.g. `datalad subdatasets`, `siblings`; so if we are to come up with a helper command (like e.g. sketched [here](https://github.com/datalad/datalad/issues/4981#issuecomment-705706777)), it should be `credentials` to stay consistent.

"When" for singular
- if we had it like `datalad.credential.<credential-id>.force-ask` then it might have become indeed more consistent with `git` having them singular. 
- If we ever to RF to establish "hierarchical commands" mimicing `git` more closely in API design, like suggested [here](https://github.com/datalad/datalad/issues/4539#issuecomment-756204514), that is where singular `datalad credential` (and thus config section) would be more appropriate.

</details>